File: models/week_raw.go
// models/week_raw.go

package models

import "cs-server/models/venue"

// WeekRawDay matches one element in the 'week_raw' array.
// This contains the full hourly busyness data (day_raw) and summary data (day_info).
type WeekRawDay struct {
	DayRaw   []int          `json:"day_raw"`
	DayInt   int            `json:"day_int"`
	DayInfo  *venue.DayInfo `json:"day_info"`
}

// WeekRawAnalysis is the explicit definition of the 'analysis' block.
type WeekRawAnalysis struct {
	WeekRaw []WeekRawDay `json:"week_raw"`
}

// RawWindow matches the 'window' block, describing the time scope of the forecast.
type RawWindow struct {
	TimeWindowStart    int    `json:"time_window_start"`
	TimeWindowStart12H string `json:"time_window_start_12h"`
	DayWindowStartInt  int    `json:"day_window_start_int"`
	DayWindowStartTxt  string `json:"day_window_start_txt"`
	DayWindowEndInt    int    `json:"day_window_end_int"`
	DayWindowEndTxt    string `json:"day_window_end_txt"`
	TimeWindowEnd      int    `json:"time_window_end"`
	TimeWindowEnd12H   string `json:"time_window_end_12h"`
	WeekWindow         string `json:"week_window"`
}

// WeekRawResponse is the top-level JSON returned by GET /forecasts/week/raw2
type WeekRawResponse struct {
	VenueAddress string          `json:"venue_address"`
	Window       RawWindow       `json:"window"`
	Status       string          `json:"status"`
	Analysis     WeekRawAnalysis `json:"analysis"` // Using the NAMED struct to avoid mock errors
	VenueName    string          `json:"venue_name"`
	VenueID      string          `json:"venue_id"`
}

File: app/models/week_raw.py
"""Weekly raw forecast data models using Pydantic."""
from typing import Optional
from pydantic import BaseModel
from app.models.venue import DayInfo


class WeekRawDay(BaseModel):
    """Single day in the weekly forecast with hourly busyness data."""
    day_raw: list[int]  # 24 hourly values
    day_int: int  # 0=Monday to 6=Sunday
    day_info: Optional[DayInfo] = None


class WeekRawAnalysis(BaseModel):
    """Analysis block containing 7 days of weekly forecast data."""
    week_raw: list[WeekRawDay]


class RawWindow(BaseModel):
    """Time window metadata describing the forecast scope."""
    time_window_start: int = 0
    time_window_start_12h: str = ""
    day_window_start_int: int = 0
    day_window_start_txt: str = ""
    day_window_end_int: int = 0
    day_window_end_txt: str = ""
    time_window_end: int = 0
    time_window_end_12h: str = ""
    week_window: str = ""


class WeekRawResponse(BaseModel):
    """Response from GET /forecasts/week/raw2 endpoint."""
    venue_address: str = ""
    window: RawWindow
    status: str
    analysis: WeekRawAnalysis
    venue_name: str = ""
    venue_id: str = ""
